utils/wrapper.js

```
@@ -0,0 +1,20 @@
+const logger = require('../common/logger')
+
+function functionWrapper(fn, fnName) {
+  return async function () {
+    try {
+      const span = await logger.startSpan(fnName ?? fn.name)
+      const result = await fn.apply(this, arguments);
+      await logger.endSpan(span)
+      return result
+    } catch (e) {
+      await logger.endSpanWithError(span, err)
+      return e
```

Isn't it expected to `throw` the error instead of simply returning it?